src/VisualStudio/Core/Def/Implementation/PullMemberUp/MainDialog/PullMemberUpViewModel.cs

```diff
@@ -0,0 +1,209 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Windows;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Utilities;
+using static Microsoft.VisualStudio.LanguageServices.Implementation.ExtractInterface.ExtractInterfaceDialogViewModel;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.PullMemberUp
+{
+    internal class PullMemberUpViewModel : AbstractNotifyPropertyChanged
+    {
+        public ObservableCollection<PullUpMemberSymbolView> SelectedMembersContainer { get; set; }
+
+        public ObservableCollection<MemberSymbolViewModelGraphNode> TargetMembersContainer { get; set; }
+
+        public ImmutableDictionary<ISymbol, PullUpMemberSymbolView> SymbolToMemberViewMap { get; }
+
+        public MemberSymbolViewModelGraphNode SelectedTarget { get => _selectedTarget; set => SetProperty(ref _selectedTarget, value, nameof(SelectedTarget)); }
+
+        private MemberSymbolViewModelGraphNode _selectedTarget;
+
+        private bool _selectAllAndDeselectAllChecked;
+
+        private readonly SemanticModel _semanticModel;
+
+        private readonly Dictionary<ISymbol, IEnumerable<ISymbol>> _dependentsMap;
+
+        public bool SelectAllAndDeselectAllChecked
+        {
+            get => _selectAllAndDeselectAllChecked;
+            set
+            {
+                _selectAllAndDeselectAllChecked = value;
+                NotifyPropertyChanged($"{nameof(SelectAllAndDeselectAllChecked)}");
+            }
+        }
+
+        internal PullMemberUpViewModel(
+            SemanticModel semanticModel,
+            ISymbol selectedNodeSymbol,
+            IGlyphService glyphService)
+        {
+            var allMembers = selectedNodeSymbol.ContainingType.GetMembers().Where(
+                    member => {
+                        if (member is IMethodSymbol methodSymbol)
+                        {
+                            return methodSymbol.MethodKind == MethodKind.Ordinary;
+                        }
+                        else if (member is IFieldSymbol fieldSymbol)
+                        {
+                            return !member.IsImplicitlyDeclared;
+                        }
+                        else
+                        {
+                            return member.Kind == SymbolKind.Property || member.Kind == SymbolKind.Event;
+                        }
+                    });
+
+            var memberinitalStates = allMembers.
```

Comment:
we shoudl not have logic in the UI layer.  The UI layer shoudl just allow people to interact with teh UI and select what they want.  This should then defer to a non-ui component to do all the work.